app/cypress/support/commands.js

```diff
@@ -258,6 +258,91 @@ Cypress.Commands.add(
   }
 );
 
+Cypress.Commands.add(
+  "addProjectSummaryReport",
+  (
+    reportDueDate,
+    receivedDate,
+    comments,
+    projectSummaryReportPayment,
+    paymentNotes,
+    dateSentToCsnr
+  ) => {
+    // Extra assertion to wait for the new report to be added
+    cy.contains(/Status of Project Summary Report/i).should("be.visible");
+
+    cy.setDateInPicker("Report Due Date", reportDueDate);
+
+    if (receivedDate) {
+      cy.setDateInPicker("Received Date", receivedDate);
+    }
+
+    if (comments) {
+      cy.get('[aria-label="General Comments"]').clear().type(comments);
+      cy.get('[aria-label="General Comments"]').should("have.value", comments);
+    }
+
+    if (projectSummaryReportPayment) {
+      cy.get('[aria-label="Project Summary Report Payment"]')
+        .clear()
+        .type(projectSummaryReportPayment);
+      cy.get('[aria-label="Project Summary Report Payment"]').should(
+        "have.value",
+        projectSummaryReportPayment
+      );
+    }
+
+    if (paymentNotes) {
+      cy.get('[aria-label="Notes for the Payment"]').clear().type(paymentNotes);
+      cy.get('[aria-label="Notes for the Payment"]').should(
+        "have.value",
+        paymentNotes
+      );
+    }
+
+    if (dateSentToCsnr) {
+      cy.setDateInPicker("Date Invoice Sent to CSNR", dateSentToCsnr);
+    }
+    return cy.url().should("include", "/form/5");
+  }
+);
+
+Cypress.Commands.add(
+  "checkProjectSummaryReport",
+  (
+    reportDueDate
+    // receivedDate,
+    // comments,
+    // projectSummaryReportPayment,
+    // paymentNotes,
+    // dateSentToCsnr
+  ) => {
+    cy.findByText(/Report Due Date$/i)
+      .next()
+      .contains(reportDueDate);
+
+    // cy.findByText(/Received Date$/i)
```

Comment:
It is a mystery to me why cypress cannot find these elements when they are present in the screenshots. Any input is welcome

Reply:
Could be a dot issue on date fields where the CI browser is different from your local browser. try something like:
`.contains(/Jun(\.)? 16, 1991/i);`

Reply:
I moved away from using the extending functions from `cypress testing library` to using the core `cypress` function. A more detailed description of these issues is [here](https://docs.google.com/document/d/14JlhKw9zxnD1GNN62XD-IOn2AL_ubPrW40I1F74wxxQ/edit#heading=h.9lzly6n5lxjr)